File: workspaces-good/crates/calculator/src/root_calculator_container.rs
use crate::agent_calculator_orchestrator::{CalculatorOrchestrator, CalculatorOrchestratorDeps};
use crate::capabilities_addition_analyzer::AdditionAnalyzer;
use crate::capabilities_division_analyzer::DivisionAnalyzer;
use crate::capabilities_multiplication_analyzer::MultiplicationAnalyzer;
use crate::capabilities_subtraction_analyzer::SubtractionAnalyzer;
use calculator_shared::contract_calculator_aggregate::CalculatorAggregate;

// ─── Block 1: Struct Definition ───────────────────────────

pub struct CalculatorContainer {
    orchestrator: CalculatorOrchestrator,
}

// ─── Block 2: Wiring & Factory ────────────────────────────

impl CalculatorContainer {
    pub fn new() -> Self {
        let orchestrator = CalculatorOrchestrator::new(CalculatorOrchestratorDeps {
            addition: Box::new(AdditionAnalyzer),
            subtraction: Box::new(SubtractionAnalyzer),
            multiplication: Box::new(MultiplicationAnalyzer),
            division: Box::new(DivisionAnalyzer),
        });
        Self { orchestrator }
    }

    pub fn orchestrator(&mut self) -> &mut dyn CalculatorAggregate {
        &mut self.orchestrator
    }
}

impl Default for CalculatorContainer {
    fn default() -> Self {
        Self::new()
    }
}

File: workspaces-good/packages/root_calculator_container.ts
import { CalculatorAggregate } from "calculator-shared/src/contract_calculator_aggregate";
import {
  CalculatorOrchestrator,
  CalculatorOrchestratorDeps,
} from "./agent_calculator_orchestrator";
import { AdditionAnalyzer } from "calculator-addition/src/capabilities_addition_analyzer";
import { SubtractionAnalyzer } from "calculator-subtraction/src/capabilities_subtraction_analyzer";
import { MultiplicationAnalyzer } from "calculator-multiplication/src/capabilities_multiplication_analyzer";
import { DivisionAnalyzer } from "calculator-division/src/capabilities_division_analyzer";

// ─── Block 1: Struct Definition ───────────────────────────

export class CalculatorContainer {
  private orchestrator: CalculatorOrchestrator;

  constructor() {
    this.orchestrator = new CalculatorOrchestrator({
      addition: new AdditionAnalyzer(),
      subtraction: new SubtractionAnalyzer(),
      multiplication: new MultiplicationAnalyzer(),
      division: new DivisionAnalyzer(),
    } as CalculatorOrchestratorDeps);
  }

  getOrchestrator(): CalculatorAggregate {
    return this.orchestrator;
  }
}
